internal/provider/incident_alert_route_resource_test.go

```diff
@@ -211,6 +211,206 @@ func TestAccIncidentAlertRouteResourceConditionalConfig(t *testing.T) {
+      escalation_targets = []                                                                                                                                                          
+    }                                                                                                                                                                                  
+                                                                                                                                                                                       
+    incident_config = {                                                                                                                                                                
+      auto_decline_enabled = false                                                                                                                                                     
+      enabled              = true                                                                                                                                                      
+      condition_groups     = []                                                                                                                                                        
+      defer_time_seconds   = 300                                                                                                                                                       
+      grouping_keys        = []                                                                                                                                                        
+    }                                                                                                                                                                                  
+                                                                                                                                                                                       
+    incident_template = {                                                                                                                                                              
+      # Focus on custom fields in alphabetical order                                                                                                                                   
+      custom_fields = [                                                                                                                                                                
+       {                                                                                                                                                                          
+          custom_field_id = incident_custom_field.alpha_field1.id                                                                                                                      
```

Comment:
I don't think this'll test what you want it to - these are correctly ordered to begin with, so the change to a set makes no difference. You'll want to specify these in the opposite order, such that the field created last is set first